BUG: `_VarArray` can't handle `MatrixVar` (#1044)
scipopt/PySCIPOpt: CHANGELOG.md

> ### Fixed
> - Raised an error when an expression is used when a variable is required
> - Fixed some compile warnings
> - _VarArray can accept MatrixVariable now

Reviewer: ```suggestion
- addConsIndicator() accepts MatrixVariable as binvar
### Fixed
- Raised an error when an expression is used when a variable is required
- Fixed some compile warnings
```

Author: There are many methods like `addConsSOS1`, `addConsSOS2`, `addConsAnd`, and more that use `_VarArray`
This pr doesn't work for `addConsIndicator`

Reviewer: Okay, can we then add a passing test and mention in the changelog what works now?
